swap/peer.go

```diff
@@ -103,10 +98,15 @@ func (sp *Peer) handleEmitChequeMsg(ctx context.Context, msg interface{}) error
 
 	// TODO: check serial and balance are higher
 
-	// reset balance to zero, TODO: fix
-	sp.swap.resetBalance(sp.ID())
+	// reset balance by amount
+	// as this is done by the creditor, receiving the cheque, the amount should be negative,
+	// so that updateBalance will calculate balance + amount which result in reducing the peer's balance
+	sp.swap.resetBalance(sp.ID(), 0-int64(cheque.Honey))
 	// send confirmation
-	if err := sp.Send(ctx, &ConfirmMsg{}); err != nil {
+	confMsg := &ConfirmMsg{
+		Cheque: cheque,
+	}
+	if err := sp.Send(ctx, confMsg); err != nil {
```

Comment:
Why are we sending confirm messages? 
The function handleConfirmMsg does nothing but logging.

Reply:
That's right, good question. The question really is: 

* do we need to confirm to the issuer that the beneficiary received it's cheque?

If not, we can remove the handling and the message even.

Reply:
We asked that in the Swarm research and this is the answer:

> Q: Right now we send cheque received confirmation message. Can we remove this?
> A: We do not need this message. For one, the TCP connection handles all ACKs, and if we don’t receive that, we must have disconnected. When we reconnect we send the last cheque as part of the handshake.

So IMHO good to remove.